### PR TITLE
[Merged by Bors] - task/update aioredis

### PR DESCRIPTION
**Pull request by sburba**
Should fix a few bugs:
* Crashing when script expires from redis
* Silently ignoring all updates after receiving an invalid update
* Use a single connection for all redis pubsub for room_store
    > Previously we used one/room which means at 200 rooms we would run out of redis connections
* Shutdown properly when _any_ task fails unexpectedly

### CODE REVIEW

File: api/src/room_store/redis_room_listener.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        await con.publish(_channel_key(room_id), json.dumps(asdict(request)))
+
+    async def _keep_connection_alive(self) -> NoReturn:
+        while True:
+            await asyncio.sleep(KEEPALIVE_INTERVAL_SECS)
+            await self._pubsub.ping()
+
+    def _on_pubsub_task_finished(self, task: Task) -> None:
+        exc = task.exception() or ValueError(
+            f"{task.get_name()} finished without throwing an exception"
```
Comment (sjohnson31) [inline api/src/room_store/redis_room_listener.py:67]:
```suggestion
            f'{task.get_name()} finished without throwing an exception'
```

File: api/tests/test_room_store.py [hunk 2]
```diff
@@ -68,6 +69,24 @@ async def test_change_notifications(room_store: RoomStore) -> None:
     assert reply == [VALID_REQUEST]
 
 
+async def test_change_notification_error(
+    redis: Redis, redis_room_store: RoomStore
+) -> None:
+    """
+    Verify that errors receiving pub/sub messages bubble out to consumers
```
Comment (sjohnson31) [inline api/tests/test_room_store.py:76]:
```suggestion
    Verify that receiving pub/sub error messages bubble out to consumers
```

File: api/tests/test_room_store.py [hunk 3]
```diff
@@ -68,6 +69,24 @@ async def test_change_notifications(room_store: RoomStore) -> None:
...
+
+    This test only runs against the redis implementation because it's impossible to
+    trigger an error in the memory implementation
+    """
+
+    changes = await redis_room_store.changes(TEST_ROOM_ID)
+    sub_task = asyncio.create_task(async_collect(changes, count=1))
+
+    await redis.publish(f"channel:{TEST_ROOM_ID}", "INVALID REQUEST")
```
Comment (sjohnson31) [inline api/tests/test_room_store.py:85]:
```suggestion
    await redis.publish(f'channel:{TEST_ROOM_ID}', 'INVALID REQUEST')
```

**Review by sjohnson31 — APPROVED** [review]

**Review by sjohnson31 — APPROVED** [review]

**Comment by sburba** [comment]
bors merge

**Comment by sburba** [comment]
bors merge

**Pull request closed without merging**